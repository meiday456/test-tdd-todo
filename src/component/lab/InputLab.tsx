import React, {ReactElement} from "react";

const InputLab = (): ReactElement => {
  return (
    <div>
      <label htmlFor={"username"}>user</label>
      <input id={"username"} data-testid="username" />
      <label htmlFor="ta">user</label>
      <textarea name="description" id="ta" cols={30} rows={10}></textarea>
    </div>
  );
};
export default InputLab;
