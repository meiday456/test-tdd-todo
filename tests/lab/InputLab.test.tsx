import {render, screen} from "@testing-library/react";
import InputLab from "@/src/component/lab/InputLab";
import userEvent from "@testing-library/user-event";
describe("Input Lab", () => {
  context("query check", () => {
    it("getBy test ID", () => {
      render(<InputLab />);
      expect(screen.getByTestId("username")).toBeInTheDocument();
    });

    it("getByRole name", () => {
      render(<InputLab />);
      // const inputEls = screen.getByRole("textbox", {name: /user/i});
      const textareaEl = screen.getByLabelText("user", {
        selector: "textarea",
      });
      expect(textareaEl).toBeInTheDocument();
    });
  });

  it("type into an input field", async () => {
    render(<input defaultValue="Hello," />);
    const input = screen.getByRole("textbox");
    await userEvent.type(input, " World!");
    expect(input).toHaveValue("Hello, World!");
    expect(input).toHaveDisplayValue("Hello, World!");
  });
});
