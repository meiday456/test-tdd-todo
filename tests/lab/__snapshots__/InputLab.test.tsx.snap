// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Lab type into an input field 1`] = `
<input
  value="Hello,"
/>
`;
